File: backend/facility-service/src/main/java/com/campus/facility/service/FacilityService.java
package com.campus.facility.service;

import com.campus.facility.dto.FacilityRequestDto;
import com.campus.facility.dto.FacilityResponseDto;
import com.campus.facility.enums.FacilityType;

import java.util.List;

public interface FacilityService {

    FacilityResponseDto createFacility(FacilityRequestDto requestDto);

    List<FacilityResponseDto> getAllFacilities();

    FacilityResponseDto getFacilityById(Long id);

    FacilityResponseDto updateFacility(Long id, FacilityRequestDto requestDto);

    void deleteFacility(Long id);

    List<FacilityResponseDto> searchFacilities(
            FacilityType type,
            Integer minCapacity,
            String location
    );
}

File: frontend/src/services/facilityService.js
const BASE_URL = "http://localhost:8083/api/facilities";

export const getAllFacilities = async () => {
  const response = await fetch(BASE_URL);
  if (!response.ok) {
    throw new Error("Failed to fetch facilities");
  }
  return response.json();
};

export const getFacilityById = async (id) => {
  const response = await fetch(`${BASE_URL}/${id}`);
  if (!response.ok) {
    throw new Error("Failed to fetch facility");
  }
  return response.json();
};

export const createFacility = async (facilityData) => {
  const response = await fetch(BASE_URL, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(facilityData),
  });

  if (!response.ok) {
    throw new Error("Failed to create facility");
  }

  return response.json();
};

export const updateFacility = async (id, facilityData) => {
  const response = await fetch(`${BASE_URL}/${id}`, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(facilityData),
  });

  if (!response.ok) {
    throw new Error("Failed to update facility");
  }

  return response.json();
};

export const deleteFacility = async (id) => {
  const response = await fetch(`${BASE_URL}/${id}`, {
    method: "DELETE",
  });

  if (!response.ok) {
    throw new Error("Failed to delete facility");
  }

  return response.text();
};

export const searchFacilities = async (filters) => {
  const params = new URLSearchParams();

  if (filters.type) params.append("type", filters.type);
  if (filters.minCapacity) params.append("minCapacity", filters.minCapacity);
  if (filters.location) params.append("location", filters.location);

  const response = await fetch(`${BASE_URL}/search?${params.toString()}`);

  if (!response.ok) {
    throw new Error("Failed to search facilities");
  }

  return response.json();
};